content/docs/security/local-ratelimit.txt

```diff
@@ -0,0 +1,247 @@
+
+2. Create an HTTPRoute that limits requests to the httpbin app along the `ratelimit.example` domain. To apply the RoutePolicy that created earlier, you use the `extensionRef` filter. 
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.networking.k8s.io/v1
+   kind: HTTPRoute
+   metadata:
+     name: httpbin-ratelimit
+     namespace: httpbin
+   spec:
+     parentRefs:
+     - name: http
+       namespace: {{< reuse "docs/snippets/ns-system.md" >}}
+     hostnames:
+     - ratelimit.example
+     rules:
+     - matches:
+       - path:
+           type: PathPrefix
+           value: /
+       filters:
+       - type: ExtensionRef
+         extensionRef:
+          name: local-ratelimit
+          group: gateway.kgateway.dev
+          kind: RoutePolicy
+       backendRefs:
+       - name: httpbin
+         port: 8000
+   EOF
+   ```
+
+3. Send a request to the httpbin app. Verify that you get back a 200 HTTP response code. 
+   {{< tabs items="LoadBalancer IP address or hostname,Port-forward for local testing" >}}
+   {{% tab  %}}
+   ```sh
+   curl -vik http://$INGRESS_GW_ADDRESS:8080/status/200 -H "host: ratelimit.example:8080"
+   ```
+   {{% /tab %}}
+   {{% tab %}}
+   ```sh
+   curl -vik localhost:8080/status/200 -H "host: ratelimit.example"
+   ```
+   {{% /tab %}}
+   {{< /tabs >}}
+   
+   Example output: 
+   ```
+   
+   ```
+   
+4. Send another request to the httpbin app. Note that this time the request is denied with a 429 HTTP response code and a `local_rate_limited` message in your CLI output. Because the route is configured with only 1 token that is refilled every 100 seconds, the token was assigned to the connection of the first request. No tokens were available to be assigned to the second request. If you wait for 100 seconds, the token bucket is refilled and a new connection can be accepted by the route. 
+   {{< tabs items="LoadBalancer IP address or hostname,Port-forward for local testing" >}}
+   {{% tab  %}}
+   ```sh
+   curl -vik http://$INGRESS_GW_ADDRESS:8080/status/200 -H "host: ratelimit.example:8080"
+   ```
+   {{% /tab %}}
+   {{% tab %}}
+   ```sh
+   curl -vik localhost:8080/status/200 -H "host: ratelimit.example"
+   ```
+   {{% /tab %}}
+   {{< /tabs >}}
+
+   Example output:
+   ```
+   ...
+   * Mark bundle as not supporting multiuse
+   < HTTP/1.1 429 Too Many Requests
+   HTTP/1.1 429 Too Many Requests
+   < x-ratelimit-limit: 1
+   x-ratelimit-limit: 1
+   < x-ratelimit-remaining: 0
+   x-ratelimit-remaining: 0
+   < x-ratelimit-reset: 79
+   x-ratelimit-reset: 79
+   ...
+   Connection #0 to host 34.XXX.XX.XXX left intact
+   local_rate_limited      
+   ```
+   
+## Gateway configuration {#gateway}
+
+Instead of applying local rate limiting to a particular route, you can also apply it to an entire gateway. This way, the local rate limiting settings are applied to all the routes that the gateway serves. 
```

Comment:
So when I have both the route and gateway, how do those work together? If either is hit, is it denied? or do I get to hit the RL for both (so I'd get 2?)